src/app/api/activity/activityLogger.service.ts

```diff
@@ -0,0 +1,124 @@
+import { ActivityType, AssigneeType, Task } from '@prisma/client'
+import { BaseService } from '@/app/api/core/services/base.service'
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import User from '../core/models/User.model'
+import { ClientResponse, CompanyResponse, InternalUsers, MeResponse } from '@/types/common'
+import { UpdateTaskRequest } from '@/types/dto/tasks.dto'
+import WorkflowStatesService from '../workflow-states/workflowStates.service'
+
+export class ActivityLogger extends BaseService {
+  public taskId: string
+
+  constructor({ taskId, user }: { taskId: string; user: User }) {
+    super(user)
+    this.taskId = taskId
+  }
+
+  async createTaskLog() {
+    const userInfo = await this.getUserInfo(this.user.token)
+
+    await this.db.activityLog.create({
+      data: {
+        taskId: this.taskId,
+        workspaceId: this.user.workspaceId,
+        activityType: ActivityType.CREATE_TASK,
+        details: {
+          initiator: userInfo?.givenName || '' + ' ' + userInfo?.familyName || '',
+          initiatorId: this.user.internalUserId as string,
+        },
+      },
+    })
+  }
+
+  async createAssignLog(payload: UpdateTaskRequest) {
+    if (payload.assigneeId) {
+      const userInfo = await this.getUserInfo(this.user.token)
+
+      const clientInfo: unknown = await this.getUserInfoById(payload, this.user.token)
+
+      let assignedTo =
+        payload.assigneeType === AssigneeType.company
+          ? (clientInfo as CompanyResponse).name
+          : `${(clientInfo as InternalUsers)?.givenName} ${(clientInfo as InternalUsers)?.familyName}`
+
+      await this.db.activityLog.create({
+        data: {
+          taskId: this.taskId,
+          workspaceId: this.user.workspaceId,
+          activityType: ActivityType.ASSIGN_TASK,
+          details: {
+            initiator: userInfo?.givenName + ' ' + userInfo?.familyName,
```

Comment:
need to trim

Reply:
Updated in efa5a82f8039ee965bcca6338125c61e15a076a9 commit